README.md

```diff
@@ -1 +1,63 @@
+# Dashboard-chart-db-explorer
+This is a pure Python framework aims to visualize data (stored in sql database) with an online dashboard. It helps define a dashboard with a set of sql queries and their associated definition for the output chart, and allows reusing without having to rewrite the program.   
+
+At present, the framework mainly applies Dash, whose links to its tutorials and other references are attached at the end of this file. 
+
+## Table of contents
+* [Functional Design](#Functional-Design)
+* [Algorithmic Design](#Algorithmic-Design)
+* [Reference](#Reference)
+
+## Functional Design
+#### Input Definition 
+
+User that requests the web dashboard need to prepare a `.JSON` file that contains layout (of the dashboard), queries and their corresponding chart_def, or other elements like markdown. By default, the x y mapping is in the order of the selected element in the corresponding query, But the users can also choose to write x y mapping explicitly. Here, the index(idx) corresponds to the order (start from 0) of the selected elements in the given query. For example, if the query select id and average_vote, then {...,"x":[1],...} means that we take average_vote as the x axis.   
+
+An example `input.json` file is presented below,
+```
+{"0chart": [{"query": "...", "chart_def": "table" (, "x": [...(index e.g 0)],"y" : [...])}, ...],
+ "1chart": [{"query": "...", "chart_def": "table" (, "x": [...],"y" : [...])}, ...],
+ "2chart": [{"query": "...", "chart_def": "table" (, "x": [...],"y" : [...])}, ...],
+ "3markdown": ["...", "..."],
+ "4markdown": ["..."],
+ ...
+ "layout": {"row0": {"4": [width, height], ...},"row1": {...}}
+}
+```
+Here, every field for chart and markdown begin with an unique int (e.g. __0__ in "0chart") as an identity number that can be used to refer to it in the layout part. Every field is considered as a section. It can has several components, which will be presented in sequential order on the page. For example, in `0chart` section, users can include two or more queries, and the result charts will be shown vertically.   
```

Comment:
I think we can allow the chart identifier to be an arbitrary string (i.e. doesn't necessarily have to start with a number) that the user can use to reference in the "layout" section of the input json file.